test/test_common/environment.cc

```diff
@@ -149,6 +149,11 @@ std::string TestEnvironment::temporaryFileSubstitute(const std::string& path,
                                                      const ParamMap& param_map,
                                                      const PortMap& port_map,
                                                      Network::Address::IpVersion version) {
+  // Protobufs should be generated dynamically so not need port substitution.
```

Comment:
This implies we'll never do templated protobuf generation. If you feel this is true as this is where you see the integration tests going medium term, I'm good with that, but otherwise it would be more flexible to allow substitutions as done for JSON today.